src/bot/channel-connection.ts

```diff
@@ -0,0 +1,107 @@
+import { gc } from "bun";
+import { getChatHandler } from "handlers";
+import { chatterTimeHandler } from "handlers/activity-handler/chatters-time";
+import { setDefaultCommandsForChannel } from "services/commands";
+import { ChannelTimer } from "services/timers";
+import { TwitchApi } from "services/twitch-api";
+import { TwitchIrc } from "services/twitch-irc";
+import { logger } from "utils/logger";
+
+type TArgs = {
+  ircClient: WebSocket;
+  channelName: `#${string}`;
+  ownerId: number;
+  authToken: string;
+};
+
+export class ChannelConnection {
+  private _api: TwitchApi;
+  private _irc: TwitchIrc;
+  private channelName: string;
+
+  private chattersInterval: Timer | undefined;
+
+  private automsgInterval: Timer | undefined;
+  private automsgTimer: ChannelTimer | undefined;
+
+  constructor(args: TArgs) {
+    logger.info(`Creating channel connection for channel: ${args.channelName}`);
+    this._irc = new TwitchIrc(args.ircClient, args.channelName, args.ownerId);
+    this._api = new TwitchApi(args.channelName, args.authToken);
+    this.channelName = args.channelName;
+  }
+
+  private async automsgChecker(): Promise<void> {
+    const info = await this.api.getStreamInfo();
+    if (!info) {
+      this.automsgTimer?.stop();
+      this.automsgTimer = undefined;
+    } else {
+      this.automsgTimer = new ChannelTimer(this.channelName, (msg) => {
```

Comment:
it creates `ChannelTimer` is created every time it runs through check, add another if to prevent that